src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -1418,6 +1441,33 @@ class ContentSidebar extends PureComponent<Props, State> {
         });
     }, DEFAULT_COLLAB_DEBOUNCE);
 
+    /**
+     * Refreshes sidebar when classification is changed
+     *
+     * @private
+     * @return {void}
+     */
+    onClassificationChange = (): void => {
+        const { fileId } = this.props;
+        if (!fileId) {
+            return;
+        }
+
+        this.fetchFile(fileId, true);
+    };
+
+    /**
+     * Add classification click handler
+     *
+     * @private
+     * @return {void}
+     */
+    onClassificationClick = (): void => {
+        const { onClassificationClick = noop } = this.props.detailsSidebarProps;
+
```

Comment:
Did you try setting isFileLoading here? Might be a better user experience if the user immediately sees the loading UI as a result of their action.